In [ ]:
### Logistic Regression B

In [5]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.mlab as mlab

from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix, accuracy_score

df = pd.read_csv("data-iris.csv")
df.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [7]:
x = df[['sepal_length','sepal_width','petal_length','petal_width']]
y = df['species']

In [10]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.25, random_state=5)

model = LogisticRegression(multi_class='multinomial')
model.fit(x_train, y_train)
predictions = model.predict(x_test)


In [15]:
print('Coffecient:', model.coef_)
print('Intercept:', model.intercept_)
print('Classificatin Report:', classification_report(y_test, predictions))
print('Accuracy Score:', accuracy_score(y_test, predictions))

Coffecient: [[-0.40433838  0.87056027 -2.2725731  -1.00750209]
 [ 0.63845996 -0.45089648 -0.17061467 -0.94682463]
 [-0.23412157 -0.41966378  2.44318778  1.95432672]]
Intercept: [  8.99355166   1.59201026 -10.58556191]
Classificatin Report:               precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        12
  versicolor       1.00      0.93      0.96        14
   virginica       0.92      1.00      0.96        12

    accuracy                           0.97        38
   macro avg       0.97      0.98      0.97        38
weighted avg       0.98      0.97      0.97        38

Accuracy Score: 0.9736842105263158


##### So model can predict 97.37% accurately

#### Prediction

###### Let's make a prediction with a new data. Suppose we have an iris flower but we do not know what species it is. Our task in this model will be to input the size of the flower sepal_length, petal_length, petal_length, petal_width in the model, and the work of the model will be to guess which species the flower is.

In [16]:
new_observation = [[4.6, 3.1, 2.9,0.2]]
model.predict(new_observation)

array(['setosa'], dtype=object)

#### Model Says that it is Setosa